### PR TITLE
Add a url parameter to specify single-threaded mode

### PR DESCRIPTION
**Pull request by chrishtr**
This is useful for comparing performance with threaded mode.

### CODE REVIEW

File: src/lab12.py [hunk 1]
```diff
@@ -1361,10 +1361,21 @@ def handle_quit(self):
 
 if __name__ == "__main__":
     import sys
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Chapter 12 code')
+    parser.add_argument("--url", default=2, type=str, required=True,
```
Comment (pavpanchekha) [inline src/lab12.py:1367]:
```suggestion
    parser.add_argument("url", type=str, required=True,
```

The name should not have dashes if it's a positional (not named) argument, and `default` describes a default value, not the position.

File: src/lab12.py [hunk 2]
```diff
@@ -1361,10 +1361,21 @@ def handle_quit(self):
...
+        help="URL to load")
+    parser.add_argument('--single_threaded', type=bool, default=False,
+        help='Whether to run the browser without a browser thread')
```
Comment (pavpanchekha) [inline src/lab12.py:1370]: I think instead of `type=bool` you want `action="store_true"`. Argparse is pretty annoying.

**Review by pavpanchekha — COMMENTED** [review]
I approve the idea of making a parameter here (we can describe it in the text if we want, or not if it's just for testing) but `argparse` is pretty screwy and I left some comments on what command line API I think you wanted. You can merge once you're happy with how the arguments are passed.

File: src/lab12.py (hunk 1 above)
Comment (chrishtr) [reply src/lab12.py]: Apparently the -- is needed...seems to work as desired now.

Examples here use it: https://docs.python.org/3/library/argparse.html

**Pull request merged**